llama_index/vector_stores/deeplake.py

```diff
@@ -211,38 +164,34 @@ def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
             ref_doc_id (str): The doc_id of the document to delete.
 
         """
-        view = None
-        if ref_doc_id:
-            view = self.ds.filter(lambda x: x["ids"].numpy().tolist() == [ref_doc_id])
-            ids = list(view.sample_indices)
-
-        with self.ds:
-            for id in sorted(ids)[::-1]:
-                self.ds.pop(id)
-
-            self.ds.commit(f"deleted {len(ids)} samples", allow_empty=True)
-            self.ds.summary()
+        self.vectorstore.delete(ids=[ref_doc_id])
```

Comment:
to be honest, I'm not sure this delete function was ever working properly.

The ref_doc_id comes from `node.ref_doc_id`, but here it seems like we are deleting based on `result.id` (which is the same as `node.node_id`)

So when a user deletes using ref_doc_id, nothing will ever get deleted 🤔

Reply:
`result.node.ref_doc_id` is different than the current `result.id` that is being stored

So if a user tried to delete using `ref_doc_id`, nothing will actually be deleted properly because right now it's keying of the field that stores `result.id`

Reply:
I see, so `ref_doc_id` is more metadata related thing right? We just need to find element that has `{"doc_id": ref_doc_id}` in metadata

Reply:
Adressed, now we are doing metadata based search